internal/services/storage/storage_account_resource.go

```diff
@@ -1453,10 +1457,13 @@ func resourceStorageAccountUpdate(d *pluginsdk.ResourceData, meta interface{}) e
 			return fmt.Errorf("Unable to locate Storage Account %q!", storageAccountName)
 		}
 
-		queueClient, err := storageClient.QueuesClient(ctx, *account)
+		// Currently, only the data plane API of queues fully support the exported properties of the `queue_properties`.
+		// TODO @magodo: once the mgmt plane API catches up, switch following client to the mgmt plane.
+		queuesDataPlaneClient, err := storageClient.QueuesDataPlaneClient(ctx, *account)
 		if err != nil {
 			return fmt.Errorf("building Queues Client: %s", err)
 		}
+		queueClient := shim.NewDataPlaneStorageQueueWrapper(queuesDataPlaneClient).(shim.DataPlaneStorageQueueWrapper)
```

Comment:
the intention of the shim layer is to avoid these implementation details within the resources, so we shouldn't be casting these inside the resources - does the `storageClient.QueuesDataPlaneClient` method not have a method defined on the interface?

Reply:
The PR description has the detailed clarification on this: 

> 3. This PR changes the interface definition of the StorageQueuesWrapper to remove the UpdateServiceProperties() and GetServiceProperties(). The reason is that these two methods don't technically belong to the queue resource, but to the queue service. The only place that is using these two methods are in the azurerm_storage_account. Currently, due to the mgmt API of the queue service lacks of the minuteMetrics, hourMetrics and the logging, we have to keep using the data plane API. However, once this gap disappears, we can use the mgmt API of the queue service, just as what has been done for the file service and the blob service in the implementation of the azurerm_storage_account. Until then, we can further remove these two methods from the DataPlaneStorageQueueWrapper.

Reply:
so if we enable Resource Manager with these type assertions this'll panic - for now we'll need to leave this in the interface (and raise an error if we attempt to enable this for Resource Manager), or use the Data Plane API here directly?

Reply:
Even if we enable the resource manager, the type assertion here won't panic because we are instantiating the `DataPlaneStorageQueueWrapper` as a `StorageQueuesWrapper` interface and casting this interface back to `DataPlaneStorageQueueWrapper`.

While you're right, we can just use the data plane client directly here.